### DISCUSSION TITLE
Development Help from Nabu Casa?

### DISCUSSION THREAD

**Discussion by GizmoGoody** [flz/iaqualink-py#56] [Q&A] [answered]
@flz Thank you for maintaining this library and HA integration. If they just kept working in their current state, I think most of us would remain grateful.

Having said that, I see that five of the six issues (I just added the sixth recently) currently posted for this library are tagged "help wanted."

I have also posted a feature request [here](https://community.home-assistant.io/t/iaqualink-device-registry-update/751937) regarding device registration. I'm curious if you think I should resubmit that as a bug as one user suggested.

But since the HA integration is in Core, is there any way for you to request additional help from the Nabu Casa team if no one in the community (presumably with an iAquaLink system and coding skills) can assist?

**Comment by flz** [comment, marked as answer]
Hey! I doubt I'll be able to get help from Nabu Casa. The integration is used by a very small number of people (409 reported installations according to the analytics [page](https://analytics.home-assistant.io/integrations/)) and their time is better on improving core (as in, essential) frameworks, features and integrations.

There is a thread in the HA forums where iaqualink gets discussed (the one I originally created, back when it was a custom component. Most people are home automation enthusiasts with no, or limited, python programming abilities.

As I've commented on many issues before, I would be thrilled to accept code contributions. My access to hardware is limited to my own setup (iaqua system) and my spare time to work on the code is pretty sporadic.

I appreciate your contributions to existing issues, hopefully someone with the time and skills can use the information to help improve the existing code.

**Reply by GizmoGoody** [reply]
Thanks. Would it help to have a few users who could test on hardware that you don't have - supplying logs and feedback to assist in development of new features?

What are your thoughts on issues like the one I posted about the device registry? It seems that would have more general implications with less dependency on hardware and testing.

**Reply by jkosharek** [reply]
I would be willing to supply logs and feedback to assist in development of new features.

**Reply by flz** [reply]
I've been wondering about this for a while. In the past, my main issue has been access to hardware and testing. At this point, I think this can be worked around by a combination of things:
* Access to users willing to try development branches or experimental patches
* Real easy way for people to do the above

I think the first one could be addressed by creating a discussion per system type and letting people subscribe for future updates and calls for testing. I think I will do that for now and link as part of the iaqualink HA documentation page.

The second one has been partly addressed by adding a cli tool to this repository (validates discovery and some basic actions) but HA testing requires more involved interaction. There's some basic documentation available on how to do this but I can make it better.

**Reply by flz** [reply]
I've updated the wiki documentation with some testing instructions and created per-system discussions. Closing this discussion and thanks for the initial suggestion!
